bmwqemu.pm

```diff
@@ -108,6 +108,21 @@ our $testedversion;
 
 sub init {
     load_vars();
+    {
+        # DEBUG in vars.json can be a single value or a list (array) of values.
+        # it's turned into a hash which should be tested using exists, like so:
+        # if (exists get_var("DEBUG")->{flag}) { ... }
+        # this allows to give a combination of flags
+        my %DEBUG_SET;
+        if (exists $vars{DEBUG}) {
+            my $DEBUG = \$vars{DEBUG};
+            if (ref $DEBUG eq "SCALAR") {
+                $DEBUG = [$$DEBUG];
+            }
+            @DEBUG_SET{@$$DEBUG} = ();
+        }
+        $vars{DEBUG} = \%DEBUG_SET;
+    };
     $vars{NAME} ||= 'noname';
```

Comment:
I thought about this.  Then I either need support functions that put the "opt1 opt2 opt3" into a separate hash, or test for it each time (smart regex).

I dared to instead fix my patch and make it work with hashes of this form, too:

    { "foo" => undef, "bar" => undef, "baz" => undef }